solr/test-framework/src/java/org/apache/solr/cloud/AbstractDistribZkTestBase.java

```diff
@@ -242,45 +240,15 @@ public static void waitForCollectionToDisappear(
     log.info("Collection has disappeared - collection:{}", collection);
   }
 
-  static void waitForNewLeader(
-      CloudSolrClient cloudClient, String shardName, Replica oldLeader, TimeOut timeOut)
+  static void waitForNewLeader(CloudSolrClient cloudClient, String shardName, Replica oldLeader)
       throws Exception {
-    log.info("Will wait for a node to become leader for {} secs", timeOut.timeLeft(SECONDS));
+    log.info("Will wait for a node to become leader for 15 secs");
     ZkStateReader zkStateReader = ZkStateReader.from(cloudClient);
-    zkStateReader.forceUpdateCollection(DEFAULT_COLLECTION);
-
-    for (; ; ) {
-      ClusterState clusterState = zkStateReader.getClusterState();
-      DocCollection coll = clusterState.getCollection("collection1");
-      Slice slice = coll.getSlice(shardName);
-      if (slice.getLeader() != null
-          && !slice.getLeader().equals(oldLeader)
-          && slice.getLeader().getState() == Replica.State.ACTIVE) {
-        if (log.isInfoEnabled()) {
-          log.info(
-              "Old leader {}, new leader {}. New leader got elected in {} ms",
-              oldLeader,
-              slice.getLeader(),
-              timeOut.timeElapsed(MILLISECONDS));
-        }
-        break;
-      }
-
-      if (timeOut.hasTimedOut()) {
```

Comment:
the loss of the diagnostics seems unfortunate; we could retain that if we time out

Reply:
Good point. But this is at the cost of potentially slower test execution since we don't unblock the test thread when we receive the ZK watch.

We would get the same if we call `logThreadDumps(...)` and `printLayoutToStream(...)` in case of error. I can do such a change.

Reply:
Good catch, thanks. I replaced the exception by a call to `fail()` with similar message.
I also removed the `InterruptedException`. This one is not supposed to happen, and that's better to not consider it in the test.